Gateway/Command/AbstractCommand.php

```diff
@@ -0,0 +1,342 @@
+        );
+
+        throw new NoSuchEntityException(
+            new Phrase(
+                sprintf("Unable to find a SwedbankPay payment matching order %s", $order->getIncrementId())
+            )
+        );
+    }
+
+    /**
+     * @param string $command
+     * @param string|int|float $amount
+     * @param OrderInterface $mageOrder
+     * @param PaymentOrderInterface|PaymentQuoteInterface $swedbankPayOrder
+     * @throws SwedbankPayException
+     */
+    protected function checkRemainingAmount($command, $amount, $mageOrder, $swedbankPayOrder)
+    {
+        $getMethod = 'getRemaining' . ucfirst($this->cmdTransActionMap[$command]) . 'Amount';
+
+        // phpcs:ignore Magento2.Functions.DiscouragedFunction
+        $remainingAmount = (int)call_user_func([$swedbankPayOrder, $getMethod]);
+
+        if ($remainingAmount >= ($amount * 100)) {
+            return;
+        }
+
+        $this->logger->error(
+            sprintf(
+                "Failed to %s order %s with SwedbankPay payment id %s:" .
+                "The amount of %s exceeds the remaining %s.",
+                $command,
+                $mageOrder->getEntityId(),
+                $swedbankPayOrder->getPaymentId(),
+                $amount,
+                $remainingAmount
+            )
+        );
+
+        throw new SwedbankPayException(
+            new Phrase(
+                sprintf(
+                    "SwedbankPay %s Error: The amount of %s exceeds the remaining %s.",
+                    ucfirst($command),
+                    $amount,
+                    $remainingAmount
+                )
+            )
+        );
+    }
+
+    /**
+     * @param string $command
+     * @param array $responseData
+     * @param OrderInterface $mageOrder
+     * @param PaymentOrderInterface|PaymentQuoteInterface $swedbankPayOrder
+     * @return string
+     * @throws SwedbankPayException
+     */
+    protected function getTransactionResult($command, $responseData, $mageOrder, $swedbankPayOrder)
+    {
+        $state = isset($responseData[$this->cmdTransActionMap[$command]]['transaction']['state']) ?
+            $responseData[$this->cmdTransActionMap[$command]]['transaction']['state'] : 'Failed';
+
+        switch ($state) {
+            case "Initialized":
+            case "AwaitingActivity":
+                $status = "pending";
+                break;
+            case "Completed":
+                $status = "complete";
+                break;
+            case "Failed":
+            default:
+                $this->logger->error(
+                    sprintf(
+                        "Failed to %s order %s with SwedbankPay payment order id %s, response data:\n%s",
+                        $command,
+                        $mageOrder->getEntityId(),
+                        $swedbankPayOrder->getPaymentId(),
+                        // phpcs:ignore Magento2.Functions.DiscouragedFunction
+                        print_r($responseData, true)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function print_r() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-magento2-payments/pullRequest?prid=5056328)